node/src/components/deploy_acceptor/tests.rs

```diff
@@ -0,0 +1,584 @@
+            | TestScenario::FromClientMissingAccount
+            | TestScenario::FromClientInvalidDeploy => false,
+        }
+    }
+}
+
+struct Reactor {
+    storage: Storage,
+    deploy_acceptor: DeployAcceptor,
+    _storage_tempdir: TempDir,
+    test_scenario: TestScenario,
+}
+
+impl reactor::Reactor for Reactor {
+    type Event = Event;
+    type Config = TestScenario;
+    type Error = Error;
+
+    fn new(
+        config: Self::Config,
+        _registry: &Registry,
+        _event_queue: EventQueueHandle<Self::Event>,
+        _rng: &mut NodeRng,
+    ) -> Result<(Self, Effects<Self::Event>), Self::Error> {
+        let (storage_config, storage_tempdir) = storage::Config::default_for_tests();
+        let storage_withdir = WithDir::new(storage_tempdir.path(), storage_config);
+        let storage = Storage::new(
+            &storage_withdir,
+            None,
+            ProtocolVersion::from_parts(1, 0, 0),
+            false,
+            "test",
+        )
+        .unwrap();
+
+        let deploy_acceptor = DeployAcceptor::new(
+            super::Config::new(true),
+            &Chainspec::from_resources("local"),
+        );
+
+        let reactor = Reactor {
+            storage,
+            deploy_acceptor,
+            _storage_tempdir: storage_tempdir,
+            test_scenario: config,
+        };
+
+        let effects = Effects::new();
+
+        Ok((reactor, effects))
+    }
+
+    fn dispatch_event(
+        &mut self,
+        effect_builder: EffectBuilder<Self::Event>,
+        rng: &mut NodeRng,
+        event: Event,
+    ) -> Effects<Self::Event> {
+        match event {
+            Event::Storage(event) => reactor::wrap_effects(
+                Event::Storage,
+                self.storage.handle_event(effect_builder, rng, event),
+            ),
+            Event::DeployAcceptor(event) => reactor::wrap_effects(
+                Event::DeployAcceptor,
+                self.deploy_acceptor
+                    .handle_event(effect_builder, rng, event),
+            ),
+            Event::ControlAnnouncement(ctrl_ann) => {
+                panic!("unhandled control announcement: {}", ctrl_ann)
+            }
+            Event::DeployAcceptorAnnouncement(_) => {
+                // We do not care about deploy acceptor announcements in the acceptor tests.
+                Effects::new()
+            }
+            Event::ContractRuntime(event) => {
+                // Contract runtime requests should not be made in the case of a deploy sent
+                // by a peer.
+                assert!(!self.test_scenario.is_from_peer());
+                match event {
+                    ContractRuntimeRequest::Query {
+                        query_request,
+                        responder,
+                    } => {
+                        let query_result =
+                            if self.test_scenario == TestScenario::FromClientMissingAccount {
```

Comment:
There are several similar comments below :)

I think we should keep such logic within the `impl` of `TestScenario`, like so:
```
    fn query_result(&self, query_request: &QueryRequest) -> QueryResult {
        match self {
            TestScenario::FromClientMissingAccount => QueryResult::ValueNotFound(String::new()),
            TestScenario::FromPeerInvalidDeploy
            | TestScenario::FromPeerValidDeploy
            | TestScenario::FromPeerRepeatedValidDeploy
            | TestScenario::FromPeerRegression
            | TestScenario::FromClientInvalidDeploy
            | TestScenario::FromClientInsufficientBalance
            | TestScenario::FromClientValidDeploy
            | TestScenario::FromClientRepeatedValidDeploy => {
                if let Key::Account(account_hash) = query_request.key() {
                    let preset_account =
                        Account::create(account_hash, BTreeMap::new(), URef::default());
                    QueryResult::Success {
                        value: Box::new(StoredValue::Account(preset_account)),
                        proofs: vec![],
                    }
                } else {
                    panic!("expect only queries using Key::Account variant");
                }
            }
        }
    }
```

and just use it on the caller side:
`let query_result = self.test_scenario.query_result(&query_request);`

Reply:
No, I don't want to extract this into a `TestScenario` cause we are mocking behavior of the reactor, and we should have it present in the implementation of the trait itself

Reply:
I'm in agreement with Karan here, although I wouldn't have objected if it were implemented in `TestScenario` either.  I slightly prefer just having it inline in the test here as I think it makes the test more understandable.